R/http.R

```diff
@@ -406,12 +406,9 @@ signatureHeaders <- function(authInfo, method, path, file = NULL) {
   # generate date
   date <- rfc2616Date()
 
-  # generate contents hash as hex values.
-  md5 <- fileMD5(file)
-
   if (!is.null(authInfo$secret)) {
     # the content hash is a string of hex characters when using secret.
-    md5 <- md5.as.string(md5)
+    md5 <- fileMD5(file)
```

Comment:
Confirmed that this change still performs at most one MD5 pass (expensive for large files).